nautobot/extras/models/change_logging.py

```diff
@@ -213,18 +213,32 @@ def get_snapshots(self):
 
         prior_change = self.get_prev_change(only=["object_data_v2", "object_data"])
 
-        if self.action != ObjectChangeActionChoices.ACTION_CREATE and prior_change is not None:
-            prechange = prior_change.object_data_v2
-            if prechange is None:
-                prechange = prior_change.object_data
+        # Populate the prechange field, create actions do not need to have a prechange field
+        if self.action != ObjectChangeActionChoices.ACTION_CREATE:
+            # Deal with the cases where we are trying to capture an object deletion/update and there is no prior change record.
+            # This can happen when the object is first created and the changelog for that object creation action is deleted.
+            if prior_change is None:
+                if self.action == ObjectChangeActionChoices.ACTION_DELETE:
+                    prechange = self.object_data_v2
+                    if prechange is None:
+                        prechange = self.object_data
+                else:
+                    prechange = pre_object_data_v2
+                    if prechange is None:
+                        prechange = pre_object_data
+            else:
+                prechange = prior_change.object_data_v2
+                if prechange is None:
+                    prechange = prior_change.object_data
 
+        # Populate the postchange field, delete actions do not need to have a postchange field
         if self.action != ObjectChangeActionChoices.ACTION_DELETE:
             postchange = self.object_data_v2
             if postchange is None:
                 postchange = self.object_data
 
         if prechange and postchange:
-            if self.object_data_v2 is None or prior_change.object_data_v2 is None:
+            if self.object_data_v2 is None or (prior_change is not None and prior_change.object_data_v2 is None):
                 prechange = prior_change.object_data
                 postchange = self.object_data
```

Comment:
What purpose does this block serve, actually? Seems weird to me that we're going to the trouble of setting `prechange` and `postchange` above, then overwriting them here.

Reply:
~I think it was needed in the pre-refactored logic?~

Looking at the old file, it feels like it was not needed in that block either. Will remove

Reply:
After some investigation, it seems that this code block was added from this [commit](https://github.com/nautobot/nautobot/commit/ca2b08ae9301c679912b67ddd48e4c590842cb50).